File: greatfrontend/1266_minimum_time_visiting_all_points.py
class Solution:
    def minTimeToVisitAllPoints(self, points: List[List[int]]) -> int:
        total_time = 0

        for i in range(len(points) - 1):
            curr_x, curr_y = points[i]
            target_x, target_y = points[i + 1]
            total_time += max(abs(target_x - curr_x), abs(target_y - curr_y))

        return total_time

File: greatfrontend/1266_minimum_time_visiting_all_points.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var minTimeToVisitAllPoints = function (points) {
  let totalTime = 0;

  // Loop through the array and find the abs diferrence between the current point and the next point
  // We'll move diagonally until we lined up in a direction
  // Then we move horizontally or vertically the remaining distance
  for (i = 0; i < points.length - 1; i++) {
    let [currX, currY] = points[i];
    let [targetX, targetY] = points[i + 1];
    // To the total time we add either the max value of the difference between the x coordinates or the y coordinates since we can move diagonally.
    totalTime += Math.max(Math.abs(targetX - currX), Math.abs(targetY - currY));
  }

  return totalTime;
};
